.github/workflows/rust.yml

```diff
@@ -53,6 +73,16 @@ jobs:
         # the binary will be used by the next cargo test step
         run: cargo build
       - name: Run cargo tests
-        run: cargo test --all-targets --all-features
+        env:
+          GITHUB_ACTIONS_BASH_CONTAINER: 1
+        run: |
+          docker run --volume $PWD:/root --network host --detach --name test_container ${{ matrix.bash_docker_image }} sleep infinity
+          # coreutils includes mktemp
+          docker exec test_container apk add coreutils curl
+          if [ "${{ matrix.sed }}" == "GNU sed" ]; then
+            docker exec test_container apk add sed
+          fi
+          cargo test --all-targets --all-features
```

Comment:
i understand that you have created a docker container to run tests inside it. this command is run at the host machine, therefore rendering the container useless

Reply:
The container is for just running bash scripts and returning the outputs, not the whole rust tests process. I've made cargo run the bash scripts inside the pre-started container: https://github.com/amber-lang/amber/blob/42551103e540d9ddf88fe0ff0bb0571e1e50d924/src/compiler.rs#L328-L329

```mermaid
flowchart LR
  subgraph "Host machine"
    cargo[cargo run]
    http[tiny_http]
    cargo -- "scripts" -->bash
    bash -- "outputs" --> cargo
    subgraph "Container"
        bash
    end
  end
```

Reply:
> The container is for just running bash scripts and returning the outputs, not the whole rust tests process

why? is there a reason to not run the whole thing in the container? thats just making an unnecessary mess out of it.